### PR TITLE
Adapt code for use with modern EM.

### PR DESCRIPTION
**Pull request by alex-oleshkevich**
No noticeable breaks found, code adopted for use with EM 3.
Also changed autoloader to PSR-4 to make it more constistent with ZF.

Solves https://github.com/zendframework/ZendDeveloperTools/issues/207

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,15 +22,15 @@
...
-        "zendframework/zend-mvc":            "2.*",
-        "zendframework/zend-eventmanager":   "2.*",
-        "zendframework/zend-stdlib":         "2.*",
-        "zendframework/zend-servicemanager": "2.*",
-        "zendframework/zend-modulemanager":  "2.*",
-        "zendframework/zend-version":        "2.*",
-        "zendframework/zend-debug":          "2.*",
-        "zendframework/zend-view":           "2.*"
+        "php":                               "^5.5 || ^7.0",
+        "zendframework/zend-mvc":            "^2.7",
```
Comment (samsonasik) [inline composer.json]: use `"^2.7 || ^3.0.0@dev"` in favor of https://github.com/zendframework/ZendSkeletonApplication/pull/329 to avoid this error:
```
$ composer update
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Loading composer repositories with package information
Updating dependencies (including require-dev)                             
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Can only install one of: zendframework/zend-mvc[2.7.x-dev, 3.0.x-dev].
    - Can only install one of: zendframework/zend-mvc[2.7.x-dev, 3.0.x-dev].
    - zendframework/zend-developer-tools dev-zf3 requires zendframework/zend-mvc ^2.7 -> satisfiable by zendframework/zend-mvc[2.7.x-dev].
    - Installation request for zendframework/zend-developer-tools dev-zf3 -> satisfiable by zendframework/zend-developer-tools[dev-zf3].
    - Installation request for zendframework/zend-mvc ^3.0.0@dev -> satisfiable by zendframework/zend-mvc[3.0.x-dev].
```

**Comment by alex-oleshkevich** [comment]
Changed

**Comment by samsonasik** [comment]
tested with zend-mvc ^3.0.0@dev used https://github.com/zendframework/ZendSkeletonApplication/pull/329 :+1: 
For zend-version usage, just in case we won't use it again, it may rely on `Application\Module::VERSION` ..

but for now, @Ocramius I think it ready for merge with may be create new tag /cc @weierophinney

**Comment by samsonasik** [comment]
@weierophinney @ocramius this needs to be merged if https://github.com/zendframework/ZendSkeletonApplication/pull/329 merged ;)

**Comment by samsonasik** [comment]
as https://github.com/zendframework/ZendSkeletonApplication/pull/329 merged,  I think it merge-able ;). The ZendDeveloperTools will support v2 and v3 after it merged

File: composer.json [hunk 2]
```diff
@@ -22,15 +22,15 @@
...
-        "zendframework/zend-mvc":            "2.*",
-        "zendframework/zend-eventmanager":   "2.*",
-        "zendframework/zend-stdlib":         "2.*",
-        "zendframework/zend-servicemanager": "2.*",
-        "zendframework/zend-modulemanager":  "2.*",
-        "zendframework/zend-version":        "2.*",
-        "zendframework/zend-debug":          "2.*",
-        "zendframework/zend-view":           "2.*"
+        "php":                               "^5.5 || ^7.0",
+        "zendframework/zend-mvc":            "^2.7 || ^3.0.0@dev",
```
Comment (samsonasik) [inline composer.json]: should ready to use: `"^2.7 || ^3.0.0",`

File: src/ZendDeveloperTools/Module.php [hunk 3]
```diff
@@ -213,4 +217,19 @@ public function getServiceConfig()
             ),
         );
     }
+    
+    /**
+     * 
+     * @return array
+     */
+    public function getAutoloaderConfig()
```
Comment (froschdesign) [inline src/ZendDeveloperTools/Module.php]: @alex-oleshkevich 
Why did you add this method?

**Comment by aight8** [comment]
Any updates?

File: src/ZendDeveloperTools/Module.php (hunk 3 above)
Comment (alex-oleshkevich) [inline src/ZendDeveloperTools/Module.php]: actually, added it to make it look closer standard module. can be removed if required.

**Comment by alex-oleshkevich** [comment]
@froschdesign, what updates do you wait from me?

**Comment by samsonasik** [comment]
ping @weierophinney @ocramius

**Comment by froschdesign** [comment]
@alex-oleshkevich 
Please remove `ZendDeveloperTools\Module::getAutoloaderConfig()`.

**Comment by samsonasik** [comment]
@froschdesign I think you can doing that on merge in case @alex-oleshkevich doesn't have a chance to update it /cc @Ocramius @weierophinney :)

**Comment by samsonasik** [comment]
@alex-oleshkevich I cherry-picked your commits in new PR #213 with include I and @froschdesign feedbacks patches, hope will get merged soon.

//cc @Ocramius @weierophinney

**Comment by alex-oleshkevich** [comment]
ok. then i close that PR as commits moved to another one.
